app/src/main/java/com/example/travellens/fragments/MessageFragment.java

```diff
@@ -2,52 +2,116 @@
 
 import android.os.Bundle;
 
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
 import androidx.fragment.app.Fragment;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
 
+import android.util.Log;
 import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
 
+import com.example.travellens.Message;
+import com.example.travellens.MyFirebaseUser;
 import com.example.travellens.R;
+import com.example.travellens.UserAdapter;
+import com.google.android.libraries.places.widget.AutocompleteSupportFragment;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseUser;
+import com.google.firebase.database.DataSnapshot;
+import com.google.firebase.database.DatabaseError;
+import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.FirebaseDatabase;
+import com.google.firebase.database.ValueEventListener;
+import com.parse.ParseUser;
 
-/**
- * A simple {@link Fragment} subclass.
- * Use the {@link MessageFragment#newInstance} factory method to
- * create an instance of this fragment.
- */
-public class MessageFragment extends Fragment {
-    private static final String ARG_PARAM1 = "param1";
-    private static final String ARG_PARAM2 = "param2";
+import java.sql.Struct;
+import java.util.ArrayList;
+import java.util.List;
 
-    private String mParam1;
-    private String mParam2;
+public class MessageFragment extends Fragment {
+    private RecyclerView recyclerView;
+    private UserAdapter userAdapter;
+    private List<MyFirebaseUser> firebaseUsers;
+    FirebaseUser user;
+    private List<String> userList;
+    DatabaseReference reference;
```

Comment:
What's reference? :)

Reply:
Reference is used for the Firebase database. It allows me to get the objects in the database reference at a specific object location. It also listens for changes in the database so it can automatically update messaging without me needing to query.